### PR TITLE
Fix: Rebuffering issues

### PR DESCRIPTION
**Pull request by realeyes-craig**
- Update Conviva to  4.0.13
- Update Bitmovin Player to 2.64.0
- Add tracking `isPlaybackStarted` to not report unintended buffering and then a playing state before playback has started
- Add more sources for testing

@luckygoyal-bitmovin I'm opening this PR with some of your changes as well because they seem valuable going forward.

### CODE REVIEW

File: BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift [hunk 1]
```diff
@@ -34,6 +34,7 @@ public final class ConvivaAnalytics: NSObject {
     let playerHelper: BitmovinPlayerHelper
     // Workaround for player issue when onPlay is sent while player is stalled
     var isStalled: Bool = false
+    var isPlaybackStarted: Bool = false
```
Comment (luckygoyal-bitmovin) [inline BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift:37]: This needs to be reset for next playback. `internalInitializeSession` and `internalEndSession` can be good places to reset this to false.

File: Example/BitmovinConvivaAnalytics/ViewController.swift [hunk 2]
```diff
@@ -82,26 +82,81 @@ class ViewController: UIViewController {
...
+        sourceItem.itemTitle = "Art of Motion"
+        // set start offset
+        let  options: SourceOptions = SourceOptions()
+        options.startOffset = 30
+        options.startOffsetTimelineReference = .start
+        sourceItem.options = options
+        return sourceItem
+    }
+
+    var liveSourceItem: SourceItem {
```
Comment (luckygoyal-bitmovin) [inline Example/BitmovinConvivaAnalytics/ViewController.swift:129]: Please remove the live source items. The included HLS live URL need not be publicly available. I can also do this before merge.

File: Example/BitmovinConvivaAnalytics/ViewController.swift [hunk 3]
```diff
@@ -82,26 +82,81 @@ class ViewController: UIViewController {
...
+            sourceString = streamString
+        }
+
+        let sourceItem = SourceItem(url: URL(string: sourceString)!)!
+        sourceItem.itemTitle = "HLS Live"
+
+        return sourceItem
+    }
+
+    var liveSourceItemStartOffset: SourceItem {
```
Comment (luckygoyal-bitmovin) [inline Example/BitmovinConvivaAnalytics/ViewController.swift:142]: Please remove the live source items. The included HLS live URL need not be publicly available. I can also do this before merge.

File: BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift (hunk 1 above)
Comment (realeyes-craig) [reply BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift:37]: Removed this

File: Example/BitmovinConvivaAnalytics/ViewController.swift (hunk 2 above)
Comment (realeyes-craig) [reply Example/BitmovinConvivaAnalytics/ViewController.swift]: Removed this

File: Example/BitmovinConvivaAnalytics/ViewController.swift (hunk 3 above)
Comment (realeyes-craig) [reply Example/BitmovinConvivaAnalytics/ViewController.swift]: Removed this

File: BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift (hunk 1 above)
Comment (realeyes-craig) [reply BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift:37]: Reset this in internalEndSession

File: BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift [hunk 4]
```diff
@@ -307,6 +308,7 @@ public final class ConvivaAnalytics: NSObject {
         sessionKey = NO_SESSION_KEY
         contentMetadataBuilder.reset()
         logger.debugLog(message: "Session ended")
+        isPlaybackStarted = false
```
Comment (luckygoyal-bitmovin) [inline BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift:311]: Just a nit. Better to announce "Session ended" as last statement, after cleaning everything:-)

**Review by luckygoyal-bitmovin — COMMENTED** [review]
Can you please share test result for following scenarios. Is the session as seen in touchstone consistent in all the below cases now?
- VOD + user play
- VOD + user play + startOffset
- VOD + autoPlay
- VOD + autoPlay + startOffset
- VOD + autoPlay + pause during preloading(before onPlaying)
- LIVE + user play
- LIVE + user play + startOffset
- LIVE + autoPlay
- LIVE + autoPlay + startOffset
- LIVE + autoPlay + pause during preloading(before onPlaying)

File: BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift (hunk 4 above)
Comment (realeyes-craig) [reply BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift:310]: That is a valuable nit. Updated.

**Comment by realeyes-craig** [comment]
> Can you please share test result for following scenarios. Is the session as seen in touchstone consistent in all the below cases now?
> 
> * VOD + user play
> * VOD + user play + startOffset
> * VOD + autoPlay
> * VOD + autoPlay + startOffset
> * VOD + autoPlay + pause during preloading(before onPlaying)
> * LIVE + user play
> * LIVE + user play + startOffset
> * LIVE + autoPlay
> * LIVE + autoPlay + startOffset
> * LIVE + autoPlay + pause during preloading(before onPlaying)

Yes but those results will have to be delivered next week.

**Comment by luckygoyal-bitmovin** [comment]
> > Can you please share test result for following scenarios. Is the session as seen in touchstone consistent in all the below cases now?
> > 
> > * VOD + user play
> > * VOD + user play + startOffset
> > * VOD + autoPlay
> > * VOD + autoPlay + startOffset
> > * VOD + autoPlay + pause during preloading(before onPlaying)
> > * LIVE + user play
> > * LIVE + user play + startOffset
> > * LIVE + autoPlay
> > * LIVE + autoPlay + startOffset
> > * LIVE + autoPlay + pause during preloading(before onPlaying)
> 
> Yes but those results will have to be delivered next week.

Hi @realeyes-craig , any update, please?

**Comment by realeyes-craig** [comment]
> > > Can you please share test result for following scenarios. Is the session as seen in touchstone consistent in all the below cases now?
> > > 
> > > * VOD + user play
> > > * VOD + user play + startOffset
> > > * VOD + autoPlay
> > > * VOD + autoPlay + startOffset
> > > * VOD + autoPlay + pause during preloading(before onPlaying)
> > > * LIVE + user play
> > > * LIVE + user play + startOffset
> > > * LIVE + autoPlay
> > > * LIVE + autoPlay + startOffset
> > > * LIVE + autoPlay + pause during preloading(before onPlaying)
> > 
> > 
> > Yes but those results will have to be delivered next week.
> 
> Hi @realeyes-craig , any update, please?

@luckygoyal-bitmovin 

Attached are the test results.

I am unable to test the following because it does not seem like there is a way to pause during preloading:
 * VOD + autoPlay + pause during preloading(before onPlaying)
 * LIVE + autoPlay + pause during preloading(before onPlaying)

In the test results you can see there is a second `onStallStarted + onStallEnded` during autoplay which happens after `onPlaying`. So this PR should address calling an onPlaying state for the first `onStallStarted + onStallEnded`
[bm-rebuffering.txt](https://github.com/bitmovin/bitmovin-player-ios-analytics-conviva/files/6736524/bm-rebuffering.txt)

**Comment by luckygoyal-bitmovin** [comment]
Thanks @realeyes-craig for sharing test results.

**Review by luckygoyal-bitmovin — APPROVED** [review]

File: Example/BitmovinConvivaAnalytics/ViewController.swift [hunk 5]
```diff
@@ -31,12 +31,12 @@ class ViewController: UIViewController {
...
         setupBitmovinPlayer()
 
-        if let posterUrl = sourceItem.posterSource {
+        /*if let posterUrl = sourceItem.posterSource {
             // Be aware that this will be executed synchronously on the main thread (change to SDWebImage if needed)
             if let data = try? Data(contentsOf: posterUrl) {
                 posterImageView.image = UIImage(data: data)
             }
-        }
+        }*/
```
Comment (stonko1994) [inline Example/BitmovinConvivaAnalytics/ViewController.swift:39]: Why commenting this out? 🤔 Please either remove it or keep it 🙂

File: Example/BitmovinConvivaAnalytics/ViewController.swift [hunk 6]
```diff
@@ -82,23 +82,45 @@ class ViewController: UIViewController {
 
     var playerConfiguration: PlayerConfiguration {
         let playerConfiguration = PlayerConfiguration()
-        playerConfiguration.sourceItem = sourceItem
+        playerConfiguration.sourceItem = vodSourceItem
+//        playerConfiguration.sourceItem = vodSourceItemStartOffset
```
Comment (stonko1994) [inline Example/BitmovinConvivaAnalytics/ViewController.swift:86]: Would it make sense to maybe provide a UI to switch between those sources? 🤔 like a TableView or any other select option to choose? 🤔 (definitely a NIT and not really necessary just thinking about a convenient way to test both use-cases without the need to re-building the app)

File: Example/BitmovinConvivaAnalytics/ViewController.swift [hunk 7]
```diff
@@ -82,23 +82,45 @@ class ViewController: UIViewController {
...
         if let streamString = streamUrlTextField.text,
            URL(string: streamString) != nil {
             sourceString = streamString
         }
 
         let sourceItem = SourceItem(url: URL(string: sourceString)!)!
-        sourceItem.posterSource = URL(string: "https://bitmovin-a.akamaihd.net/content/poster/hd/RedBull.jpg")
         sourceItem.itemTitle = "Art of Motion"
+        // set start offset
+        let  options: SourceOptions = SourceOptions()
```
Comment (stonko1994) [inline Example/BitmovinConvivaAnalytics/ViewController.swift:120]:
```suggestion
        let options: SourceOptions = SourceOptions()
```

File: BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift (hunk 1 above)
Comment (stonko1994) [inline BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift:37]: In the `MetadataBuilder` we already have the same concept but we called the property `var playbackStarted: Bool = false` there.

I would propose to use the same naming here 🙂 _And making it private_

```suggestion
    private var playbackStarted: Bool = false
```

File: Example/BitmovinConvivaAnalytics/ViewController.swift (hunk 5 above)
Comment (luckygoyal-bitmovin) [reply Example/BitmovinConvivaAnalytics/ViewController.swift]: Uncommented the code.

File: Example/BitmovinConvivaAnalytics/ViewController.swift (hunk 6 above)
Comment (luckygoyal-bitmovin) [reply Example/BitmovinConvivaAnalytics/ViewController.swift:86]: Good input but keeping it as it is for now.

File: Example/BitmovinConvivaAnalytics/ViewController.swift (hunk 7 above)
Comment (luckygoyal-bitmovin) [reply Example/BitmovinConvivaAnalytics/ViewController.swift]: done

File: BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift (hunk 1 above)
Comment (luckygoyal-bitmovin) [reply BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift]: done

**Comment by luckygoyal-bitmovin** [comment]
@realeyes-craig , I have taken the liberty to address the review comments from @stonko1994.

**Review by stonko1994 — APPROVED** [review]

**Pull request merged**